### ISSUE TITLE
Should have instructions for building without Eclipse

### ISSUE THREAD

**Issue by wyatt8740** [sussman/twisty#72]
I don't like IDE's, and while I was able to get a lot of code building with ant and the 'ndk-build' tool, it falters on roboglk and I have absolutely no idea how to compile its code without an IDE because no information was provided. I am away from my computer anyway and doing all of this building over SSH, so needing a full blown IDE and X server is not ideal.

Any tips?

**Comment by sussman** [comment]
I'm a hardcore terminal + emacs coder myself -- been writing C and C++ that way for 25 years.  :-)

BUT:  building Android apps without an IDE is considered total insanity by the entire Android world.  Android and Java are both so ridiculously complex that it's nearly impossible to replace them with a simple Makefile... especially since the IDE contains tons of specialized tools for UI creation and testing.  (iOS is in the same boat -- impossible to do *anything* without Apple's IDE.)

If you look at the public docs on developer.android.com, it seems like you can invoke Gradle (the 'make'-like system) directly from the commandline:  http://developer.android.com/tools/building/building-cmdline.html

Good luck with that, if you're up for it.  :-)

All that said:  even the current BUILD instructions are out of date.  In 2015 Google switched from their own fork of Eclipse to their own fork of IntelliJ.   If you download the latest Android IDE, you'll have IntelliJ, and nobody's yet bothered to make this whole thing compile yet.  Theoretically IntelliJ has a migration tool that can read former Eclipse projects, but it's kind of sketchy and I haven't done it yet.  That's the *real* work that needs to happen here, in order to keep developing this project.

**Comment by wyatt8740** [comment]
I tried gradle, but I still have no idea what I'm doing because it is not aware of the existance of roboglk (there's no project related files directly in there that describe how to build it.

Thanks for the fast response, I'll keep at it. If you have any idea what linking roboglk with eclipse actually is doing, I'd love to hear it. I  am very familiar with C, but only vaguely with java (took one college credit class in High School for it. I am more of a hobbyist programmer).

**Comment by sussman** [comment]
Right. First someone has to make IntelliJ convert the Eclipse project to
IntelliJ format. That will cause the gradle files to exist. Then you can
build with gradle on the command line.

**Comment by wyatt8740** [comment]
hm, so no way to do it over SSH. dang. I'll try next time I have a normal PC in front of me.

**Comment by MaxHex** [comment]
When will it be ready for the alpha stage

**Comment by wyatt8740** [comment]
Still not gotten time to really mangle with it. I have gotten it to build, but still need to do the actual brunt of the work. I currently have a long and low-paying (non programming-related) job so I can't really spend a lot of time on this anymore :(
